app/models/agents/email_digest_agent.rb

```diff
@@ -37,18 +37,16 @@ def working?
     end
 
     def receive(incoming_events)
+      self.memory['events'] ||= []
       incoming_events.each do |event|
-        self.memory['queue'] ||= []
-        self.memory['queue'] << event.payload
-        self.memory['events'] ||= []
         self.memory['events'] << event.id
       end
     end
 
     def check
-      if self.memory['queue'] && self.memory['queue'].length > 0
-        ids = self.memory['events'].join(",")
-        groups = self.memory['queue'].map { |payload| present(payload) }
+      if self.memory['events'] && self.memory['events'].length > 0
+        payloads = received_events.unscoped.where(id: self.memory['events']).pluck(:payload).to_a
```

Comment:
The `received_events` has a default scope ordering that goes against what the specs wanted. The default scope make newest first, and the specs for the email body were expecting oldest first. I'd love a way to not fight default scoping